File: app/src/main/java/com/scatl/uestcbbs/module/post/adapter/CreateCommentImageAdapter.java
package com.scatl.uestcbbs.module.post.adapter;

import android.content.Context;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;

/**
 * author: sca_tl
 * description:
 * date: 2019/8/17 15:35
 */
public class CreateCommentImageAdapter extends BaseQuickAdapter<String, BaseViewHolder> {

    public CreateCommentImageAdapter(int layoutResId) {
        super(layoutResId);
    }

    public void delete(int position) {
        getData().remove(position);
        notifyItemRemoved(position);
        notifyDataSetChanged();
    }

    @Override
    protected void convert(BaseViewHolder helper, String item) {
        Glide.with(mContext).load(item).into((ImageView) helper.getView(R.id.item_post_create_comment_img));
        helper.addOnClickListener(R.id.item_post_create_comment_deleta_img);
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/post/adapter/CreateCommentImageAdapter.kt
package com.scatl.uestcbbs.module.post.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemCreateCommentImageBinding
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/4/20 16:12
 */
class CreateCommentImageAdapter : PreloadAdapter<String, ItemCreateCommentImageBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemCreateCommentImageBinding {
        return ItemCreateCommentImageBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemCreateCommentImageBinding>, position: Int, item: String?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.image.load(item)
    }
}